zerver/views/registration.py

```diff
@@ -76,8 +76,12 @@ def check_prereg_key_and_redirect(request: HttpRequest, confirmation_key: str) -
 
 @require_post
 def accounts_register(request: HttpRequest) -> HttpResponse:
-    key = request.POST['key']
-    confirmation = Confirmation.objects.get(confirmation_key=key)
+    try:
+        key = request.POST['key']
+        confirmation = Confirmation.objects.get(confirmation_key=key)
+    except Confirmation.DoesNotExist:
+        return render(request, "zerver/confirmation_link_expired_error.html")
```

Comment:
This commit isn't correct right now, because as of this commit `confirmation_link_expired_error.html` doesn't yet exist. That's why I suggested doing this as a follow-up commit (so as the last on), since that's the easiest way to handle that.

Reply:
aaaah ok. So I open another PR after this one get merged?

Reply:
Nono, I thikn you can simply move this commit to be the last one in the sequence.